### PR TITLE
SAML SSO

### PR DESCRIPTION
**Pull request by EmilyNecciai**
# Pull Request/Issue Resolution

**Description of Change:**

**Reason for Change:**

Closes #**ISSUE_NUMBER_HERE**

---
---

## PR Checklist
- [ ] Ensure you have completed our CLA.
- [ ] Tag @EmilyNecciai as a reviewer when the your work is done and ready to be reviewed for merge. 
- [ ] Consult the [Docs Text Formatting Guide](https://github.com/Appboy/success/wiki/Docs-Text-Formatting-Guide) to be sure you're utilizing the proper markdown formatting.
- [ ] Consult the [Docs Writing Style Guide & Best Practices](https://github.com/Appboy/success/wiki/Writing-Style-Guide-&-Best-Practices) to be sure you're aligning with our voice and other style best practices.
- [ ] [Preview your deployed changes](https://homeslice.braze.com/docs) to confirm that none of your changes break production. Pay close attention to links and images.
- [ ] Tag others as Reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `config/nginx.conf.erb`.

---
---

### CODE REVIEW

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+Sign-On URL: `https://<SUBDOMAIN>.braze.com/sign_in`
+
+Assertion Consumer Service (ACS) URL: `https://<SUBDOMAIN>/auth/saml/callback`
+For some IdPs, this can also be referred to as the Reply URL, Entity ID, Audience URL, or Audience URI.
+
+Enter the appropriate domain based on your cluster.
+
+In addition, you’ll need to setup SAML attribute mapping.
+
+| SAML Attribute | Required? | Schema |
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on.md]: need to include all options for the fields. the URL is actually a specific type in addition to the others listed. 

image

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+|---|---|---|
+|`emailAddress` | Required | `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/email` |
+| `first_name` | Optional | `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/first_name` |
+| `last_name` | Optional | `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/last_name` |
+
+{% alert note %}
+Braze only requires `emailAddress` in the SAML Assertion.
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on.md]: update emailAddress to `email`

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+{% endalert %}
+
+Once you have setup Braze within your IdP, the IdP will provide a Target URL and x.509 certificate which you will input into the Braze dashboard.
+
+After your Customer Success Manager has enabled SAML SSO for your account, you should Company Settings > Security Settings and toggle the SAML SSO section on.
+On this page, you will be asked to input:
+
+SAML Name: This will appear as the button text on the login screen. This is typically your IdP name, like “Okta.”
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on.md]: do we want to include a screenshot here of the developer settings?

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+If you want your dashboard users to only sign in with SAML SSO, you can restrict single sign-on authentication within Company Settings.
+
+{% endtab %}
+{% tab Identity Provider %}
+### Service Provider (SP) Initiated Login
+
+To enable IdP initiated login, you will first need to create an API key in Developer Settings > API Settings.
+
+Input the generated API Key as the RelayState parameter within your IdP which will be used to identity which company the user is trying to log into.
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on.md]: should we put a screenshot here for developer console > create api key?

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+Certificate: The x.509 certificate is provided by your IdP.
+
+When you save your Security Settings and logout, you should now be able to sign in with your IdP.
+
+If you want your dashboard users to only sign in with SAML SSO, you can restrict single sign-on authentication within Company Settings.
+
+{% endtab %}
+{% tab Identity Provider %}
+### Service Provider (SP) Initiated Login
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on.md]: Should be Identity Provider (IdP) Initiated Login

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
+---
+nav_title: SAML & Single Sign On
+page_order: 5
+layout: featured
+guide_top_header: "SAML & Single Sign On"
+guide_top_text: "Do you want to ensure a secure login for your company's Braze users through a company you're already using? Well, you're in luck! Braze has an integration for that!"
+
+guide_featured_title: "Which secure sign on option would you like to use?"
```
[the remark this answers is not in the document]
Comment (dkim32) [reply _docs/_user_guide/administrative/access_braze/single_sign_on.md]: strange. links aren't working for the IdPs.

File: _docs/_user_guide/administrative/access_braze/single_sign_on/okta.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ nav_title: Okta
 page_order: 1
 ---
 
-# Okta Single Sign On
+# Okta IdP
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on/okta.md]: Remove IdP

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (EmilyNecciai) [reply _docs/_user_guide/administrative/access_braze/single_sign_on.md]: I think you're editing a previous version - I'll push out the new one. Unfortunately, I don't think I can move the buttons down. I can revise it, though. I'll push an update in a few minutes - let me know what you think of it.

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md (hunk 4 above)
Comment (EmilyNecciai) [reply _docs/_user_guide/administrative/access_braze/single_sign_on.md]: I'd love to actually put in a gif of the whole process - is there a braze account I can use to mimic it?

File: _docs/_user_guide/administrative/access_braze/single_sign_on/set_up.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+|---|---|---|
+|`email` | Required | `email` <br> `mail` <br> `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/email` |
+| `first_name` | Optional | `first_name` <br> `firstname` <br> `firstName`<br>`http://schemas.xmlsoap.org/ws/2005/05/identity/claims/first_name` |
+| `last_name` | Optional | `last_name` <br> `lastname` <br> `lastName` <br>`http://schemas.xmlsoap.org/ws/2005/05/identity/claims/last_name` |
+
+{% alert note %}
+Braze only requires `email` in the SAML Assertion.
+{% endalert %}
+
+### Create and Enable a Braze API Key for IdP Login
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on/set_up.md]: this should have a larger headline and potentially moved to the bottom. confusing that "IdP iniatiated login" setup is incorporated into "SP Initiated Login" section.

Comment (dkim32) [reply _docs/_user_guide/administrative/access_braze/single_sign_on/set_up.md]: Ah. I see - i hate nesting.. but maybe nest  the Identity Provider initiated login under SAML SSO Setup and move this section to that page. 

image

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+layout: featured
+guide_top_header: "SAML & Single Sign On"
+guide_top_text: "Single Sign-On (SSO) provides companies a secure and centralized way of controlling access to the Braze dashboard. In short, a single set of credentials can be used to access different applications, including Braze. <br> <br> Braze supports Google SSO support via Open ID Connect and SAML SSO which supports the latest Security Assertion Markup Language (SAML 2.0) industry standards."
+
+guide_featured_title: "SAML SSO Topics"
+guide_featured_list:
+- name: Custom SAML SSO Set Up
+  link: /docs/user_guide/administrative/access_braze/single-sign-on/set_up/
+  fa_icon: fas fa-fingerprint
+- name: Restrict to SAML SSO Only
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on.md]: update to "Restrict to SSO Only" as it applies to both saml sso and google.

File: _docs/_user_guide/administrative/access_braze/single_sign_on/okta.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+## Step 3: Configure Okta
+
+1. In Okta, select the `Sign On` tab for the Braze SAML app, then click `Edit`.
+2. Enter your Braze API Key value from the previous step into the `Default Relay State:` field.
+3. Click `Save`.
+
+You should now be able to log into Braze using Okta!
+
+{% alert note %}
+__SAML SSO-Only Login__
```
Comment (dkim32) [inline _docs/_user_guide/administrative/access_braze/single_sign_on/okta.md]: should we include on azure ad and onelogin as well? or remove entirely?

File: _docs/_user_guide/administrative/access_braze/single_sign_on/set_up.md (hunk 8 above)
Comment (EmilyNecciai) [reply _docs/_user_guide/administrative/access_braze/single_sign_on/set_up.md]: Oh yeah, that was a mistake - the IDP thing is gone in my newest push - 2 minutes! :)

File: _docs/_user_guide/administrative/access_braze/single_sign_on/okta.md (hunk 10 above)
Comment (EmilyNecciai) [reply _docs/_user_guide/administrative/access_braze/single_sign_on/okta.md]: Yes - I changed the wording a bit. :) Got it!

File: _docs/_user_guide/administrative/access_braze/single_sign_on.md (hunk 3 above)
Comment (EmilyNecciai) [reply _docs/_user_guide/administrative/access_braze/single_sign_on.md]: I included a gif. :)

**Pull request merged**